app/js/actions/currentdirectory.js

```diff
@@ -52,14 +52,37 @@ const setArtifactDir = path => ({
 export const selectArtifactDirectory = () => {
     return (dispatch, getState) => {
         const currPath = getState().currentDirectory;
-        remote.dialog.showOpenDialog({
-            title: 'Choose Artifact Directory or File',
-            defaultpath: currPath,
-            properties: ['openFile', 'openDirectory']
-        }, (fps) => {
-            if (fps) {
-                dispatch(setArtifactDir(fps[0]));
+        let curProps = ['openFile', 'openDirectory'];
+        let curTitle = 'Choose Artifact Directory or File';
+        const openSelectArtifactDirectoryDialog = () => {
+            remote.dialog.showOpenDialog({
+                title: curTitle,
+                defaultpath: currPath,
+                properties: curProps
+            }, (fps) => {
+                if (fps) {
+                    dispatch(setArtifactDir(fps[0]));
+                }
+            });
+        };
+        if (process.platform !== 'darwin') {
+            // linux or windows
+            const callback = remote.dialog.showMessageBox({
+                type: 'question',
+                buttons: ['File', 'Directory', 'Cancel'],
+                title: 'Artifact Selection',
+                message: 'What would you like to select?',
```

Comment:
Sorry one last comment. Change the message to: `Are you importing a File or Directory?`